packages/@aws-cdk/aws-efs/README.md

```diff
@@ -15,4 +15,59 @@
 ---
 <!--END STABILITY BANNER-->
 
+This construct library allows you to set up AWS Elastic File System (EFS).
+
+```ts
+import efs = require('@aws-cdk/aws-efs');
+
+const myVpc = new ec2.Vpc(this, 'VPC');
+const fileSystem = new efs.EfsFileSystem(this, 'MyEfsFileSystem', {
+    vpc: myVpc,
+    encrypted: true,
+    lifecyclePolicy: EfsLifecyclePolicyProperty.AFTER_14_DAYS,
+    performanceMode: EfsPerformanceMode.GENERAL_PURPOSE,
+    throughputMode: EfsThroughputMode.BURSTING
+});
+```
+
+### Connecting
+
+To control who can access the EFS, use the `.connections` attribute. EFS has
+a fixed default port, so you don't need to specify the port:
+
+```ts
+fileSystem.connections.allowDefaultPortFrom(instance);
+```
+### Mounting the file system using User Data
+
+In order to automatically mount this file system during instance launch, 
+following code can be used as reference:
+```
+const inst = new Instance(this, 'inst', {
+    instanceType: InstanceType.of(InstanceClass.T2, InstanceSize.LARGE),
+    machineImage: new AmazonLinuxImage({
+        generation: AmazonLinuxGeneration.AMAZON_LINUX_2
+    }),
+    vpc,
+    vpcSubnets: {
+        subnetType: SubnetType.PUBLIC,
+    },
+});
+
+inst.userData.addCommands("yum check-update -y",
```

Comment:
I don't like putting in commands that will fail on one of the platforms and then silently ignoring the errors. Be prepared for UserData scripts to be executed under `bash -e`.

In the example you are creating an Amazon Linux instance, so no real need to add the Ubuntu commands though. If you want to show the commands for multiple platforms (admirable) then show the difference in comments:

```ts
inst.userData.addCommands(
  'yum check-update -y',  // Ubuntu: apt-get -y update
   ...);
```

Also this example is incomplete, it doesn't show creation of the EFS filesystem and how to allow connections.